genesis-builder/framework/evm/sources/Evm.move

```diff
@@ -58,6 +70,12 @@ module Evm::evm {
         native_evm_create(caller_addr, get_asset_value(value), data)
     }
 
+    /// Encode the move value into bytes using the Solidity ABI
+    /// such that it would be suitable for passing to a Solidity contract's function.
+    /// The prefix can be used to prepend the output with a Solidity 4-byte function
+    /// selector if needed.
+    public native fun abi_encode_params<T>(prefix: vector<u8>, value: T): vector<u8>;
```

Comment:
Really nice!
Just curious about 2 things (1) Do we create a separate native impl, even though it's part of the same Evm module, so that we can isolate Move to EVM calls? (2) Is this considered the infrastructure for cross-contract calls?

Reply:
(1) It could be a separate native impl since it does not use any of the EVM machinery; it is just a pure function. But on the other hand, I'm not sure why a MoveVM instance would want this native without having the other EVM natives also. I cannot think of a reason why a Move contract would want to use this encoding without then also calling the EVM.

Having be another function in the same native is a little less boilerplate, so unless we have a particular reason to separate the encoding out, I think it is fine as-is.

(2) This is infrastructure to make it easier to call EVM contracts from Move contracts. We'll use it ourselves to call the OP stack L2 contracts. And if we open up this EVM to all users in the future then it will also be useful in general.